### DISCUSSION TITLE
Waterfall chart improvement request re: category names and colors

### DISCUSSION THREAD

**Discussion by sfirke** [apache/superset#27856] [Ideas]
The waterfall chart is a nice addition to Superset!  Two ways it could be even better, as of 3.1.1:

1. You can customize the color for each category by clicking on its color box.  But this wasn't obvious to me until I looked at screenshots from the PR that overhauled the feature.  A tool tip/info box would help make this clearer.
2. I would like to be able to change the category names.  "Increase" and "Decrease" work for me but I want my chart to say "Fund Balance" instead of "Total".  However, "Total" is hard-coded.

**Comment by eliaetzensperger** [comment]
We have a similar problem with the waterfall chart, as it cannot be customised sufficiently

**Comment by SBIN2010** [comment]
I can implement this.
<img width="321" height="243" alt="Снимок экрана от 2025-08-25 00-09-24" src="https://github.com/user-attachments/assets/7642b4d3-58be-48dd-873a-4b16b78ba247" />

This is how it turned out for me, but I would appreciate an outside perspective on how logically these settings could be called.

**Reply by eliaetzensperger** [reply]
That would be perfect! :-)

**Reply by sfirke** [reply]
@SBIN2010 that would be fantastic, thank you! Since the language we use elsewhere is `label` like `X axis label`, what do you think about calling those fields Increase label, Decrease label, and Total label?

And would you also be able to address my first item above, making it clearer to the user that they can change the color? One way would be to regroup things so that there's Increase label, Increase color, then Decrease, etc... But I'm open to whatever. And no worries if you don't want to take this part on.

Let us know if you need any support, otherwise feel free to tag me on the PR for review!

**Reply by SBIN2010** [reply]
@sfirke  Agreed Increase label, Decrease label and Total label would be clearer.
How about description, any suggestions?

**Reply by sfirke** [reply]
Can you clarify what you mean by "description"?

**Reply by SBIN2010** [reply]
I meant the tooltip

**Reply by sfirke** [reply]
Let's go with something simple like "Choose the color for each series" or "Choose the color for this series" if you decide to split the colors up to go with each one's label. Once this is at the PR stage, we can get additional feedback on small details like.

**Comment by SBIN2010** [comment]
PR created https://github.com/apache/superset/pull/34847
